inc/spbc-auth.php

```diff
@@ -486,6 +489,50 @@ function spbc_2fa__key_remove_old()
     $spbc->save('data');
 }
 
+/**
+ * Handle for check if rate limit is passed
+ *
+ * @return bool
+ */
+function spbc_2fa_is_rate_limit_pass()
+{
+    $rateLimit = get_option('spbc_rate_limit_2fa', [
+        'limit' => 10,
+        'expires_in' => time() + 60,
+        'attempts' => 0,
+    ]);
+
+    if ($rateLimit['expires_in'] <= time()) {
+        $rateLimit['expires_in'] = time() + 60;
+        $rateLimit['attempts'] = 0;
+    }
+
+    if ($rateLimit['expires_in'] > time()) {
+        $rateLimit['attempts']++;
+    }
+
+    if ($rateLimit['attempts'] >= $rateLimit['limit']) {
+        return false;
+    }
+
+    update_option('spbc_rate_limit_2fa', $rateLimit);
+
+    return true;
+}
+
+/**
+ * Check if rate limit is passed
+ */
+function spbc_2fa_rate_limit()
+{
+    if ( isset($_POST['spbc_2fa'], $_POST['log']) ) {
+        $rateLimit = spbc_2fa_is_rate_limit_pass();
+        if (!$rateLimit) {
+            wp_die('Rate limit exceeded', 'Forbidden', array('response' => 403));
```

Comment:
Please add the sign that the reason is Security by CleanTalk.

Reply:
Fixed https://github.com/CleanTalk/security-malware-firewall/pull/370/commits/0a977c9a46ec058ff7a2e6e9012d8673168e218b